loadNeustarReputationIntoES.js

```diff
@@ -0,0 +1,1288 @@
+    } else if (program.upload != null) {
+        uploadCsv();
+    } else if (program.switch != null || program.switchToLastUsedIndex != null) {
+        switchIndex();
+    } else {
+        throw new Error("doOperationBasedOnOption failed. Aborting...");
+    }
+}
+
+function deleteIndex(indexToBeDeleted, cb) {
+    var options = {
+        host: program.host,
+        port: program.port,
+        path: '/' + indexToBeDeleted + '/',
+        method: 'DELETE'
+    };
+
+    var callback = function(response) {
+        var indexDeleted = response.statusCode;
+        var str = '';
+        //another chunk of data has been recieved, so append it to `str`
+        response.on('data', function (chunk) {
+            str += chunk;
+        });
+
+        //the whole response has been recieved, so we just print it out here
+        response.on('end', function () {
+            if (indexDeleted != '200') {
+                winston.log('warn', indexToBeDeleted + "failed to be deleted with response code = " + indexDeleted);
+                winston.log('warn', "Index deletion error is " + str);
+                cb(false, indexToBeDeleted);
+            } else {
+                winston.log('info', "Unused index " + indexToBeDeleted + " deleted");
+                cb(true, indexToBeDeleted);
+            }
+        });
+    };
+
+    http.request(options, callback).end();
+}
+
+function deleteUnusedIndexes(deleteAll) {
+    assertScratchSpaceIndexExists(function (success) {
+        if (success === true) {
+            getAllNeustarIndexes(function(arrayOfIndexes) {
+                if (arrayOfIndexes == null) {
+                    winston.log('info', "No indexes to delete.");
+                    return;
+                }
+                getCurrentIndexAndBlockSizeFromMetadata(function(currentMonolithIndex, currentMaxBlockSize) {
+                    getScratchSpaceInfo(function (scratchSpaceInfo) {
+                        var loadedIndex = null;
+                        var pausedIndex = null;
+                        var lastUsedIndex = null;
+                        if (scratchSpaceInfo != null) {
+                            loadedIndex = scratchSpaceInfo.loadedIndex;
+                            pausedIndex = scratchSpaceInfo.pausedIndex;
+                            lastUsedIndex = scratchSpaceInfo.lastUsedIndex;
+                            if (loadedIndex != null) {
+                                loadedIndex = loadedIndex.trim();
+                            }
+                            if (pausedIndex != null) {
+                                pausedIndex = pausedIndex.trim();
+                            }
+                            if (lastUsedIndex != null) {
+                                lastUsedIndex = lastUsedIndex.trim();
+                            }
+                        }
+                        var somethingDeleted = false;
+                        for (var i = 0; i < arrayOfIndexes.length; i++) {
```

Comment:
use `for (let index of arrayOfIndexes) { ... }` if `i` isn't used for anything else then accessing an element